packages/composables/src/factories/createCheckoutStep.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  computed,
+  reactive,
+  watch,
+  ref,
+  toRefs,
+  Ref,
+} from "@vue/composition-api";
+import cookieUniversal from "cookie-universal";
+
+/**
+ * @alpha
+ */
+export interface CheckoutStepFields {
+  [property: string]: unknown;
+}
+
+/**
+ * @alpha
+ */
+export interface VuelidateValidation {
+  $invalid: boolean;
+  $touch: () => void;
+}
+
+/**
+ * @alpha
+ */
+export interface CreateCheckoutStep {
+  isValid: Readonly<Ref<boolean>>;
+  validations: Readonly<Ref<Readonly<VuelidateValidation> | null>>;
+  setValidations: ($v: VuelidateValidation) => void;
+  validate: () => void;
+  [property: string]: any;
+}
+
+/**
+ * @alpha
+ */
+export function createCheckoutStep({
+  stepNumber,
+  stepFields,
+  stepDataUpdated,
+}: {
+  stepNumber: number;
+  stepFields: CheckoutStepFields;
+  stepDataUpdated: (updatedData: CheckoutStepFields) => void;
+}) {
+  const cookies = cookieUniversal();
```

Comment:
in the future, we can take under consideration leaving here only the interfaces, and pass the right instance - here we have good example of it - the cookies interface with get and set.

Reply:
oh, you did it with VuelidateValidation :+1:

Reply:
we use this lib anyway for setting api-client in next, that's why I moved it here for now; in future, for sure we can do it with other tools as well :)